### ISSUE TITLE
[Core feature] Rust Data Persistence

### ISSUE THREAD

**Issue by pingsutw** [flyteorg/flyte#4081]
### Motivation: Why do you think this is important?

Write a Rust plugin to read/write s3 files. It can replace current data persistence plugin in flytekit.

### Goal: What should the final outcome look like, ideally?

The Rust plugin should offer better data read/write performance than the current one

### Describe alternatives you've considered 

Use existing data persistence plugin.

###  Propose: Link/Inline OR Additional context 

https://docs.rs/object_store/latest/object_store/

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by samhita-alla** [comment]
@pingsutw, aren't we using fsspec instead of data persistence already?

**Comment by pingsutw** [comment]
Yes, we use fsspec, but we want to have a Rust data persistence plugin to replace it. Reading data using fsspec is still very slow. @troychiu is working on it.

**Comment by kumare3** [comment]
This has been now done as part of 2.0
